src/Configuration/Types.hs

```diff
@@ -0,0 +1,383 @@
+    , ccfgKeyPrefix              :: !Text
+    , ccfgRelays                 :: !Text
+    , ccfgUpdateServer           :: !Text
+    , ccfgReportServer           :: !Text
+    , ccfgInstallDirectorySuffix :: !Text
+    , ccfgMacPackageSuffix       :: !Text
+    , ccfgWalletPort             :: !Natural
+    } deriving (Eq, Generic, Show)
+
+-- Defining each 'Intrepret' instance.
+instance Interpret ClusterConfig where
+    autoWith opt = interpretAutoWithOption $ opt {fieldModifier = lowerHead . T.drop 4}
+
+instance Inject ClusterConfig where
+    injectWith opt = injectAutoWithOption $ opt {fieldModifier = lowerHead . T.drop 4}
+
+
+instance Arbitrary ClusterConfig where
+    arbitrary = do
+        ccfgName                   <- genSafeText
+        ccfgKeyPrefix              <- genSafeText
+        ccfgRelays                 <- genSafeText
+        ccfgUpdateServer           <- genSafeText
+        ccfgReportServer           <- genSafeText
+        ccfgInstallDirectorySuffix <- genSafeText
+        ccfgMacPackageSuffix       <- genSafeText
+        ccfgWalletPort             <- arbitrary
+
+        pure ClusterConfig {..}
+
+-- | OS configuration
+data OSConfig = OSConfig
+    { osName              :: !Text
+    , osConfigurationYaml :: !Text
+    , osInstallDirectory  :: !Text
+    , osX509ToolPath      :: !Text
+    , osNodeArgs          :: !NodeArgs
+    , osPass              :: !Pass
+    } deriving (Eq, Generic, Show)
+
+instance Interpret OSConfig where
+    autoWith opt = interpretAutoWithOption $ opt {fieldModifier = lowerHead . T.drop 2}
+
+instance Inject OSConfig where
+    injectWith opt = injectAutoWithOption $ opt {fieldModifier = lowerHead . T.drop 2}
+
+instance Arbitrary OSConfig where
+    arbitrary = do
+        osName              <- elements ["windows64", "macos64", "linux64"]
+        osConfigurationYaml <- genSafeText
+        osInstallDirectory  <- genSafeText
+        osX509ToolPath      <- genSafeText
+        osNodeArgs          <- arbitrary
+        osPass              <- arbitrary
+
+        pure OSConfig{..}
+
+-- | Node arguments
+data NodeArgs = NodeArgs
+    { naKeyfile          :: !Text
+    , naLogsPrefix       :: !Text
+    , naTopology         :: !Text
+    , naUpdateLatestPath :: !Text
+    , naWalletDBPath     :: !Text
+    , naTlsPath          :: !Text
+    } deriving (Eq, Generic, Show)
+
+instance Interpret NodeArgs where
+    autoWith opt = interpretAutoWithOption $ opt {fieldModifier = lowerHead . T.drop 2}
+
+instance Inject NodeArgs where
+    injectWith opt = injectAutoWithOption $ opt {fieldModifier = lowerHead . T.drop 2}
+
+instance Arbitrary NodeArgs where
+    arbitrary = do
+        naKeyfile          <- genSafeText
+        naLogsPrefix       <- genSafeText
+        naTopology         <- genSafeText
+        naUpdateLatestPath <- genSafeText
+        naWalletDBPath     <- genSafeText
+        naTlsPath          <- genSafeText
+
+        pure NodeArgs{..}
+
+-- | Paths
+data Pass = Pass
```

Comment:
I really don't understand why it's called `Pass`. Maybe `Paths` was the intention?

Reply:
This is one of the field that was on
https://github.com/input-output-hk/daedalus/blob/develop/installers/dhall/os.type#L14
so I've just copied it. I'll check with devops about it

Reply:
I've checked with devops and told me `pass` is just an name of variable so I'll change chang to `Param`!